### PR TITLE
FOGL-5891 Documentation

### PR DESCRIPTION
**Pull request by deepanshu-yadav**

### CODE REVIEW

**Review by praveen-garg — COMMENTED** [review]
clean up the README.rst and give reference to docs/index.rst

**Review by praveen-garg — APPROVED** [review]
Minor corrections/ formatting we can take in patch PR as required.

File: docs/index.rst [hunk 1]
```diff
@@ -15,14 +17,67 @@ We can also copy the timestamp if present in the CSV file. This time stamp becom
...
+                If variable Columns is false then it indicates how columns are considered.
+
+                1. pick_from_file : The columns will be picked using a row index given.
+
+                2. explicit : Specify the columns inside useColumns parameter.
+
+
+
+   - **'autoGeneratePrefix': type: string default: 'column'**:
+               If variable Columns is set true then data points will generated using the prefix.
```
Comment (AmandeepArora) [inline docs/index.rst:79]: If "variableCols" is set to true, then...

File: docs/index.rst [hunk 2]
```diff
@@ -58,6 +113,11 @@ The plugin can also play the file in a loop which means it can start again if en
 
                 The data type will be converted to integer. Also column will be renamed.
 
+|config2|
+
+   - **'rowIndexForColumnNames': type: integer default: '0'**:
+                If column method is pick_from_file then it is the index where
```
Comment (AmandeepArora) [inline docs/index.rst:119]: "where from where"

File: docs/index.rst [hunk 3]
```diff
@@ -88,36 +148,77 @@ The plugin can also play the file in a loop which means it can start again if en
                 It is left to the user to ensure there are no missing values in CSV file. However if the option selected is report. Then plugin will check for NaN's and report error to user. This can serve as a way to check the CSV file for missing values. However the user has to take action on what to do with NaN values. The default action is to ignore them.
                 When error is reported the user must delete the south service and try again with clean CSV file.
 
-  - **'repeatLoop': type: boolean default: false**:
-                Read CSV in a loop i.e. on reaching End Of File, again go back to beginning of the file.
+   - **'postProcessMethod': type: enumeration default: 'continue_playing'**:
+                It is the method to process the CSV file once all rows are ingested.
+                It could be:
+
+                1. continue_playing
```
Comment (AmandeepArora) [inline docs/index.rst:155]: It is not really "continue_playing", it is replaying.

File: docs/index.rst [hunk 4]
```diff
@@ -88,36 +148,77 @@ The plugin can also play the file in a loop which means it can start again if en
...
+       res=$(curl -sX POST http://localhost:8081/fledge/service -d  @- << EOF | jq '.'
+       {
+        "name":"My_south",
+        "type":"south",
+        "plugin":"csvplayback",
+        "enabled":false,
+        "config": {
+             "assetName":{"value":"My_csv_asset"},
+             "csvDirName":{"value":"FLEDGE_DATA/csv_data"},
+             "csvFileName":{"value":"vib"},
```
Comment (AmandeepArora) [inline docs/index.rst:187]: csv filename is "vibration.csv" as per line 174.

File: docs/index.rst [hunk 5]
```diff
@@ -148,12 +249,12 @@ The behaviour of plugin under various modes.
...
-For using poll continuous increase the readingPerSec category to the sample rate.
+For using poll mode in continuous setting increase the readingPerSec category to the sample rate.
 
 .. code-block:: console
 
       sampling_rate=8000
       curl -sX PUT http://localhost:8081/fledge/category/My_southAdvanced -d '{"bufferThreshold":"'"$sampling_rate"'","readingsPerSec":"'"$sampling_rate"'"}' |jq
 
-It is advisable to increase the buffer threshold at least half the sample rate for good performance. (As done in command)
+It is advisable to increase the buffer threshold at least half the sample rate for good performance. (As done in above command)
```
Comment (AmandeepArora) [inline docs/index.rst:259]: ...threshold to atleast half...

File: docs/index.rst (hunk 3 above)
Comment (deepanshu-yadav) [reply docs/index.rst:155]: Will change that in future.

File: docs/index.rst (hunk 4 above)
Comment (deepanshu-yadav) [reply docs/index.rst:187]: explanation given .

Comment (AmandeepArora) [reply docs/index.rst:187]: This is a strange and confusing feature of searching files with given prefix when field name is "csvfilename". Not sure why such prefix based search is needed? And if multiple match, pick first match, this is very confusing behavior. Does user really want multiple files to be processed? If so, call the parameter to indicate the same, filename pattern or something like that.

**Review by AmandeepArora — APPROVED** [review]

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**